### PR TITLE
Use testGroup for defining test jobs

### PR DESCRIPTION
**Pull request by echesakovMSFT**
As we discussed offline with @jashook @sbomer - this is a proposed solution to avoid duplicating timeouts, priorities and scenarios when create a new test job.

Basically, this PR adds a mapping between **testGroup** string and values of the following parameters
* scenarios
* priority
* timeoutTotalInMinutes
* timeoutPerTestCollectionInMinutes
* timeoutPerTestInMinutes

and uses **testGroup** on a level of azure-pipeline.yml to set these values.

### CODE REVIEW

File: eng/test-job.yml [hunk 1]
```diff
@@ -139,18 +150,30 @@ jobs:
           helixSource: ci/dotnet/coreclr/$(Build.SourceBranch)
 
         ${{ if eq(parameters.readyToRun, false) }}:
-          helixType: ${{ format('test/functional/cli/pri{0}', parameters.priority) }}
+          helixType: 'test/functional/cli/'
```
Comment (echesakovMSFT) [inline eng/test-job.yml]: @RussKeldorph This would fix the links to mc.dot.net
 https://github.com/dotnet/core-eng/pull/5245#issuecomment-464225583

**Comment by sbomer** [comment]
LGTM, thank you!

**Pull request merged**